contrib/dist/linux/openmpi.spec

```diff
@@ -212,7 +207,14 @@
 %define __check_files %{nil}
 %endif
 
-%{!?configure_options: %define configure_options %{nil}}
+# Set this to any options you want to pass in to configure.  By
+# default, we pass in flags to tell configure to expect to find all
+# the 3rd party packages elsewhere on the system (vs. using the
+# bundled versions).  This behavior is strongly recomended
+# (vs. including other, 3rd party packages in this Open MPI RPM).
+# Meaning: if you set a value to configure_options, you are strongly
+# advised to *also* include all the options listed below.
+%{!?configure_options: %define configure_options --with-libevent=external --with-hwloc=external --with-pmix=external --with-prrte=external}
```

Comment:
I guess we don't need the requires lines, because the shared library tracing code will do that for us (although I wonder if we should to support a `--enable-static --disable-shared` case, or maybe document said case doesn't work.

It does feel like we could be more user friendly and BuildRequires the devel packages for hwloc and libevent.  But maybe that's a bridge too far?

Reply:
Fair point.  This gets a little tricky since the user is also able to define `configure_options`, but I think that this script+specfile can't protect the user from themselves in all cases.  However, I'll at least separate the two issues slightly by:

1. Keep `configure_options`, but resume its defaults back to `%{nil}` / `""`
2. Add a boolean `all_external_3rd_party` which will add `--with-libevent=external --with-hwloc=external ...` to the `configure` command line after `%{configure_options}`.

Reply:
@bwbarrett I pushed a "SQUASHME" commit up here so you can see the changes.  If you're good with it, I'll squash it back down to a single commit.